server/config/config.go

```diff
@@ -2,53 +2,18 @@ package config
 
 import (
 	"flag"
-	"fmt"
-	"os"
 
 	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
-	"gopkg.in/yaml.v3"
-
-	"github.com/buildbuddy-io/buildbuddy/server/util/log"
-)
-
-var (
-	configFile = flag.String("config_file", "/config.yaml", "The path to a buildbuddy config file")
 )
 
-func PopulateFlagsFromData(data []byte) error {
-	// expand environment variables
-	expandedData := []byte(os.ExpandEnv(string(data)))
+const pathFlagName = "config_file"
 
-	yamlMap := make(map[string]interface{})
-	if err := yaml.Unmarshal([]byte(expandedData), yamlMap); err != nil {
-		return fmt.Errorf("Error parsing config file: %s", err)
-	}
-	typeMap, err := flagutil.GenerateYAMLTypeMapFromFlags()
-	if err != nil {
-		return err
-	}
-	if err := flagutil.RetypeAndFilterYAMLMap(yamlMap, typeMap, []string{}); err != nil {
-		return fmt.Errorf("Error encountered retyping YAML map: %s", err)
-	}
+var configPath = flag.String(pathFlagName, "/config.yaml", "The path to a buildbuddy config file")
 
-	return flagutil.PopulateFlagsFromYAMLMap(yamlMap)
+func init() {
+	flagutil.IgnoreFlag(pathFlagName)
```

Comment:
nit: can you add a comment explaining why we do this?

Reply:
Done!